vimeo-networking/src/main/java/com/vimeo/networking/model/BaseResponseList.java

```diff
@@ -37,6 +49,92 @@
     public Paging paging;
     public ArrayList<T> data;
 
-    // TODO: maybe don't need
     public abstract Class getModelClass();
+
+    public static abstract class BaseResponseTypeAdapter<S extends BaseResponseList> extends TypeAdapter<S> {
+
+        @NotNull
+        private final PagingTypeAdapter mPagingTypeAdapter;
+        @NotNull
+        private final TypeAdapter mTypeAdapter;
+
+        public BaseResponseTypeAdapter(@NotNull PagingTypeAdapter pagingTypeAdapter,
+                                       @NotNull TypeAdapter typeAdapter) {
+            mPagingTypeAdapter = pagingTypeAdapter;
+            mTypeAdapter = typeAdapter;
+        }
+
+        protected abstract S createResponseListObject();
+
+        @Override
+        public void write(JsonWriter out, S value) throws IOException {
+            out.beginObject();
+            if (value == null) {
+                ClientLogger.d("BaseResponseList null in write()");
+                out.endObject();
+                return;
+            }
+
+            out.name("total").value(value.total);
+            out.name("page").value(value.page);
+            out.name("per_page").value(value.perPage);
```

Comment:
These stings should probably be pulled out into constants since they're referenced from multiple locations (with the same value). But also it seems like this has to be something we could find an annotation for. Now if we want to add a field, instead of just adding it with one line, it takes knowing to add it to the read and write methods and also pairing it with what type it is. If it's an array it looks like it's even more work.

Additionally, if we're going this route then we'd have to do this for all of models every time we create a new object which isn't very frequent, but it's also just added time.